docs/search-v2-change-indices.md

```diff
@@ -0,0 +1,128 @@
+
+1. [Get the mapping of the index in prod](https://www.elastic.co/guide/en/elasticsearch/reference/current/indices-get-mapping.html)
+2. [Get the settings of the index in prod](https://www.elastic.co/guide/en/elasticsearch/reference/current/indices-get-settings.html)
+3. [Create a new index locally](https://www.elastic.co/guide/en/elasticsearch/reference/current/indices-create-index.html),
+using parts of the above responses:
+
+    ```
+    PUT /my-index-000001
+    {
+      "mappings": {
+        < prod_index_name/_mapping["prod_index_name"]["mappings"] >
+      },
+      "settings: {
+        "analyzers": {
+          < prod_index_name/_settings["prod_index_name"]["settings"]["index"]["analysis"] >
+        }
+      }
+    }
+    ```
+
+4. Point your Document at this index and run `./manage.py es7_init --limit DocumentName`
+
+If this succeeds,
+you can be relatively confident that this is a config change which doesn't require creating a new index.
+
+## Changes appended to a current index
+
+Your approach here will completely depend on what kind of changes you're making on the index.
+
+For example,
+if you're adding a new field which hasn't existed before,
+you'll want to make sure any changes you make which read or query ElasticSearch don't assume that the field is immediately available.
+This is because,
+until you've fully reindexed the documents in this index,
+documents will exist without the field you're adding.
+
+So if you were adding a `searchme` field to a document which defaults to true,
+rather than adding a filter to only include documents where `searchme=True`,
+you would want your query to exclude documents where `searchme=False`,
+in order to avoid having a moment during your deploy where nothing appears in the query results.
+
+```eval_rst
+.. Note::
+  If you're removing a field from a Document,
+  it would be preferable to leave it in the Document class,
+  setting its value on all documents to `None`,
+  with a note to remove it the next time a change requiring a new index is made.
+
+  This is to avoid the index on prod containing fields which don't exist on indices created afresh locally.
+```
+
+## Changes requiring a new index
+
+### Development process
+
+#### Config changes
+
+Make any changes to Documents, analyzers or index configuration which you need to do as usual,
+and increment the `Document.Index.version` attribute.
+
+#### Write operations
+
+These will be run against the new index immediately,
+as the new index will be initialized at deploy time by the `bin/run-post-deploy.sh` script,
+and writes don't go through the alias.
+
+#### Read operations
+
+Place the changes behind a call to the `Document.alias_points_at_latest_version` property,
+where the prior behavior is what's run when the property returns False,
+and the new behavior is what's run when the property returns True.
+
+For instance,
+if you were adding an `all_names` field to `ProfileDocument` to search over only one field,
```

Comment:
We discussed this in the past too but the more I am thinking about this the more uncomfortable I get with such an approach. If we must have some kind of control on the queries because of the reindexing I strongly prefer a two step release for this.

Having said that, with the `simple_query_string` we get fault tolerance for and worst case if we try to query for a field while our aliases have diverged we will get no results. In most cases during a re-index the aliases should allow searching and indexing to work simultaneously just fine. If there is a case that it's not enough, I would prefer to have a SOP on how to approach that problem than dumping this to the code and/or feature flippers.